scheduler/feasible.go

```diff
@@ -216,6 +216,27 @@ func (h *HostVolumeChecker) hasVolumes(n *structs.Node) bool {
 			if !capOk {
 				return false
 			}
+			if req.Sticky {
+				// NOTE: surely there is a better way to find the right alloc?
+				// Should we perhaps search for allocs by job? Could there be a
+				// situation in which there are non-terminal allocations
+				// belonging to the job in question that are sticky, have the
+				// volume IDs that match what the node offers and should not
+				// end up in this check?
+				allocs, err := h.ctx.ProposedAllocs(n.ID)
+				if err != nil {
+					continue
+				}
+
+				for _, a := range allocs {
+					if a.TerminalStatus() || a.NodeID != n.ID {
+						continue
+					}
+
+					return slices.Contains(a.HostVolumeIDs, volCfg.ID)
```

Comment:
If we return early, we only end up checking a single allocation. But we could have multiple allocations that are mounting the same volume. So we need to return false if the slice is non-empty and doesn't contain the `volCfg.ID`, but let it fall through otherwise.